scio-core/src/main/scala/com/spotify/scio/coders/CaseClassSerializer.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.coders
+
+import com.esotericsoftware.kryo.Kryo
+import com.esotericsoftware.kryo.io.Input
+import com.esotericsoftware.kryo.serializers.FieldSerializer
+import com.spotify.scio.util.ScioUtil
+import org.objenesis.instantiator.ObjectInstantiator
+import org.objenesis.instantiator.sun.SunReflectionFactoryInstantiator
+
+import scala.reflect.ClassTag
+
+/**
+  * Optimized serializer that can be used for case classes.
+  * Has lower memory and CPU overhead than default serialization.
```

Comment:
Benchmark?

Reply:
@ravwojdyla I'll attach a benchmark in this PR. I didn't notice the presence of benchmarks around coders.

Reply:
@ravwojdyla @bilisie yeah let's do this in a different PR